parquet/file/page_reader.go

```diff
@@ -144,6 +161,13 @@ func NewDataPageV1WithStats(buffer *memory.Buffer, num int32, encoding, defEncod
 	return ret
 }
 
+// NewDataPageV1WithExtras is like WithStats but adds the first row index and size statistics
+func NewDataPageV1WithExtras(buffer *memory.Buffer, num int32, encoding, defEncoding, repEncoding parquet.Encoding, uncompressedSize int32, stats metadata.EncodedStatistics, firstRowIdx int64, szStats SizeStatistics) *DataPageV1 {
```

Comment:
Is it worth putting some of these parameters in a struct so they can be extended in the future? Or is this very likely to be the "last" extension? (Since now we have NewV1, NewV1WithStats, and NewV1WithExtras)